ts/torch_handler/base_handler.py

```diff
@@ -160,49 +163,39 @@ def initialize(self, context):
             self.model.eval()
 
         # Convert your model by following instructions: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html
-        # TODO(msaroufim): Refactor into utils https://github.com/pytorch/serve/issues/1631
-        elif self.model_pt_path.endswith(".onnx") and onnx_enabled:
-            # self.model = self._load_onnx_model(self.model_pt_path)
-            providers = (
-                ["CUDAExecutionProvider", "CPUExecutionProvider"]
-                if self.map_location == "cuda"
-                else ["CPUExecutionProvider"]
-            )
-
-            # Set the right inference options, we can add more options here depending on what people want
-            sess_options = ort.SessionOptions()
-            sess_options.intra_op_num_threads = psutil.cpu_count(logical=True)
-
-            # Start an inference session
-            ort_session = ort.InferenceSession(
-                self.model_pt_path, providers=providers, sess_options=sess_options
-            )
-            self.model = ort_session
+        elif self.model_pt_path.endswith(".onnx") and ONNX_AVAILABLE:
+            self.model = setup_ort_session(self.model_pt_path)
+            logger.info("Succesfully setup ort session")
 
         else:
             raise RuntimeError("No model weights could be loaded")
 
-        optimization_config = os.path.join(model_dir, "compile.json")
-        backend = load_compiler_config(optimization_config)
+        if hasattr(self, "model_yaml_config") and "pt2" in self.model_yaml_config:
+            pt2_backend = self.model_yaml_config["pt2"]
+            valid_backend = check_valid_pt2_backend(pt2_backend)
+        else:
+            valid_backend = False
 
         # PT 2.0 support is opt in
-        if check_pt2_enabled() and backend:
+        if PT2_AVAILABLE and valid_backend:
             # Compilation will delay your model initialization
             try:
                 self.model = torch.compile(
                     self.model,
-                    backend=backend,
-                    mode="default" if TORCHXLA_AVAILABLE else "reduce-overhead",
+                    backend=pt2_backend,
+                    mode="default" if XLA_AVAILABLE else "reduce-overhead",
```

Comment:
@msaroufim does "reduce-overhead" force the cuda graphs? in that case I am thinking if we could default to something more flexible as it may break more often.

Reply:
alternatively adding a comment/to suggest this mode/ warn about it.

Reply:
So for online inference with small batch size I feel like if we can't do reduce overhead then we shouldn't use torch.compile at all. Granted maybe we should also pass in the mode as a field in the yaml file

Reply:
agree with making it a config as I guess it increase the time/mem pressure for slightly larger models --> run into OOM easily.